src/main/java/org/verdictdb/coordinator/VerdictSingleResultFromListData.java

```diff
@@ -2,34 +2,47 @@
 
 import com.google.common.base.Optional;
 import com.rits.cloning.Cloner;
+import org.verdictdb.commons.AttributeValueRetrievalHelper;
 import org.verdictdb.connection.DataTypeConverter;
 import org.verdictdb.connection.DbmsQueryResult;
 import org.verdictdb.connection.DbmsQueryResultMetaData;
 
 import java.io.*;
 import java.sql.Types;
+import java.util.ArrayList;
 import java.util.List;
 
-public class VerdictSingleResultFromListData extends VerdictSingleResult {
+public class VerdictSingleResultFromListData
+    extends AttributeValueRetrievalHelper implements VerdictSingleResult {
 
   private Optional<List<Object>> result;
 
+  List<String> fieldsName = new ArrayList<>();
+
   // used to support wasnull()
   private Object lastValueRead;
 
   int cursor = -1;
 
-  public VerdictSingleResultFromListData(List<Object> result) {
+  public VerdictSingleResultFromListData(ExecutionContext.QueryType type, List<Object> result) {
     super();
     if (result == null) {
       this.result = Optional.absent();
     } else {
+      if (type.equals(ExecutionContext.QueryType.show_databases)) {
```

Comment:
Please move this part to the ExecutionContext. The goal is to make this class agnostic to the caller.